pkg/apis/sparkoperator.k8s.io/v1beta2/defaults.go

```diff
@@ -70,6 +70,6 @@ func setExecutorSpecDefaults(spec *ExecutorSpec, sparkConf map[string]string) {
 	}
 	if _, exists := sparkConf["spark.executor.instances"]; !exists && spec.Instances == nil {
 		spec.Instances = new(int32)
-		*spec.Instances = 1
+		*spec.Instances = 0
```

Comment:
C: I think we'd rather do something like  "only set spec.Instances to a default _if_ `dynamicAllocation` isn't configured" as opposed to changing the default to 0

Reply:
Spark will take the max value of spark.dynamicAllocation.initialExecutors, spark.dynamicAllocation.minExecutors and spark.executor.instances if dynamic allocation is enabled, so it should still have a value, and defaulting it to zero is effectively taking it out of consideration for how many executors should be spun up automatically, since that should really be derived from the dynamic allocation settings instead

Reply:
Updated